### ISSUE TITLE
scala version of dependencies.lock is incorrect in iceberg-flink

### ISSUE THREAD

**Issue by JingsongLi** [apache/iceberg#1176]
It should depend on flink with 2.12, but is flink with 2.11 in dependencies.lock.

**Comment by JingsongLi** [comment]
CC: @openinx

**Comment by openinx** [comment]
I think I made this mistake in PR 20651b1c,  I used the flink with scala 2.11 version firstly  and then we discussed to better to support scala 2.12, so I changed the flink module from scala 2.11 to scala 2.12 but forgot to re-generate the `dependencies.lock`. 

IMO, we'd better to remove those `dependencies.lock` from git and change to re-generate them when building the iceberg modules, because: 
1.  we don't have to remember to re-generate those lock files by handle after changed the dependencies jar/version;
2. Git won't inflate because of those lock changes, in the last commit 20651b1c I commited a PR with 10000+ lines which make the changed lines untrue.
